lib/auth/user_repository_impl.dart

```diff
@@ -0,0 +1,28 @@
+
+import 'dart:convert';
+import 'package:chat_app_front/auth/user.dart';
+import 'package:chat_app_front/auth/user_repository.dart';
+import 'package:chat_app_front/config.dart';
+import 'package:http/http.dart' as http;
+
+class UserRepositoryImpl implements UserRepository{
+  @override
+  Future<String> signUpUser(User user) async {
+    var url = SERVER_IP + '/user/post';
```

Comment:
Trochę lipa, że url (bo to w sumie nie jest ip swoją drogą) serwera jest wklepaną do kodu stałą zamiast być w pliku konfiguracyjnym